core/chain_makers_test.go

```diff
@@ -97,4 +101,302 @@ func ExampleGenerateChain() {
 	// balance of addr1: 989000
 	// balance of addr2: 10000
 	// balance of addr3: 19687500000000001000
+} */
+
+// Runner function
+func chainsValidator(chain []*types.Block, primeChain BlockChain, regionChain BlockChain, zoneChain BlockChain, specsPool []blockSpecs) (validNetwork [3]BlockChain) {
```

Comment:
I don't think this file is the right place for these functions. `chain_makers.go` contains logic to make chains. `chain_makers_test.go` contains logic to test the chain makers. These functions are using chainmakers to test the node under different chain fork scenarios, but not actually testing chainmakers itself.

I'll create a new file for the chain fork tests, and that file can hold the logic which uses chainmakers to test the node under different chain fork scenarios. I'll link you the PR when its done.

Reply:
Opened a PR with a new file for the chain fork tests. Please move such logic into that file, and keep `chain_makers_test.go` focused on testing the chain makers:
https://github.com/spruce-solutions/go-quai/pull/318

Reply:
Checked it, we can refactor once the generation logic is finished.